Visiblity

TheLegendaryGanjoo opened the pull request.

TheLegendaryGanjoo committed:
visibility fix

TheLegendaryGanjoo committed:
visibility corrections

signs were slipped

TheLegendaryGanjoo committed:
visiblity fixes + comments

Andallfor commented:
Changes:
1. Make visibility and raycast static classes/functions
2. Please don't use flags in that way.
   1. Your method works (?) but just like please don't. It makes your code way messier then it needs to be.
   2. Instead, think of it as binary. 000 means do not search for anything, 100 means only search for planets, 010 means only search for satellites, and 011 means search for only satellites and facilities for example. (Please do not hardcode this). This also means that you should switch the type from int to byte (or whatever the binary equivalent is).
3. Can you add a possibility to have the second position instead be a vector that extends infinitely? We will need to be able to raycast in a direction later on, rather then between two points. (Put this in a different function).
4. For if statements that contain only one thing inside (ie line 36), use the if statement shorthand. `if (inRadius < p.radius) c++;`
5. What is the purpose of line 40? `c = c + 0;`
6. A lot of the code is just copy paste. Instead of having 3 separate loops that each do effectively the same thing, combine them. Since you're getting the cartesian position of all 3 different types of objects, have an opening if statement retrieve these positions, then have the math follow it, rather then copying the math 3 times.
7. Please use shorthand increment/decrement. `x -= 100;` not `x = x - 100;`
8. Add optional parameter that specifies facility altitude.
9. What is the purpose of `defaultRadius`? You don't seem to use it anywhere in the code.
10. Add an exclusion parameter- a list of things that we do not care about if we hit.
11. Add a target parameter- only care if we hit this object or not.
12. Add function that returns what we hit, not just if we hit something or not.
   1. Note that having all of this in one function would be pretty unwieldy, so you will probably want to split the functionality up into multiple functions instead.
13. What is the purpose of `check`? Just return true/false instead of returning check.
14. Add some termination statements.
   1. If we have already hit something, just return true since this function only asks for if we hit something, not what we hit. Any other processing done after we hit something is pointless.
   2. Add a statement in the math that skips the processing if it is clear that the object has no chance of colliding with the raycast.

Request another review after you finish editing.

TheLegendaryGanjoo committed:
Revamp of Visibility

still need to do the vector-based functions, but redid flags and added custom struct

TheLegendaryGanjoo committed:
Added vector functions

Andallfor commented on Assets/Code/data classes/visibilty.cs:
```diff
@@ -3,112 +3,189 @@
+    }
+  }
+
+  public static double distance(position p1, position p2)
```
Remove, we already have this function in position.

Andallfor commented on Assets/Code/data classes/visibilty.cs:
```diff
@@ -3,112 +3,189 @@
+     return distance;
+  }
+
+  public static double raycastMath(position position1, position position2, position obj)
```
This is only used in the current file, and likely will not be used elsewhere. Make private instead of public.

Andallfor commented on Assets/Code/data classes/visibilty.cs:
```diff
@@ -3,112 +3,189 @@
+
+  }
+
+  //double.max
```
Before you submit code, please clean it up. This means removing unnecessary code and empty lines.

Andallfor commented on Assets/Code/data classes/visibilty.cs:
```diff
@@ -3,112 +3,189 @@
+
+  public static raycastInfo raycastHit(position p1, position p2, raycastParameters flags, int defaultRadius)
   {
     // 100 indicates planet search
```
Remove leftover comments.

Andallfor commented on Assets/Code/data classes/visibilty.cs:
```diff
@@ -3,112 +3,189 @@
-          c = c + 1;
-        }
-        else
+        if (distance(p1, p.pos) > distance(p1,p2))
```
You repeatedly use distance(p1, p2). Instead of recalc every if loop, save it in the beginning of the function then just reference it rather then recalc it.

Andallfor commented on Assets/Code/data classes/visibilty.cs:
```diff
@@ -3,112 +3,189 @@
+
+  //double.max
+
+  public static raycastInfo raycastHit(position p1, position p2, raycastParameters flags, int defaultRadius)
```
Name functions better. You have raycastHit which detects all things it hits, and raycast which only detects the first. "hit" does not imply everything. Additionally, add a comment (/// <summary> comment </summary>) above each function that gives a very brief overview of what it does.

Andallfor commented on Assets/Code/data classes/visibilty.cs:
```diff
@@ -188,4 +188,150 @@ public static raycastInfo raycast(position p1, position p2, raycastParameters fl
   }
+
+//vector functions, the second number is the vector
+public static raycastInfo raycastvector(position p1, position vector, raycastParameters flags, int defaultRadius)
```
Camel case.
Move comment above function into `/// <summary> text </summary>`

Andallfor commented on Assets/Code/data classes/visibilty.cs:
```diff
@@ -188,4 +188,150 @@ public static raycastInfo raycast(position p1, position p2, raycastParameters fl
+  // 10 indicates a satellite search
+  // 1 indicates a facility search
+
+  List<planet> hitPlanets = new List<planet>();
```
You already have a struct that uses those same values. Use the struct to hold this data then just return it.

Andallfor commented on Assets/Code/data classes/visibilty.cs:
```diff
@@ -188,4 +188,150 @@ public static raycastInfo raycast(position p1, position p2, raycastParameters fl
+  bool hit = false;
+
+  position p2 = new position(
+  double.MaxValue/4 + vector.x,
```
4 needs to be 4.0. Otherwise it will be read as an int and lose precision.

Andallfor reviewed (changes requested):
Some of the comments from the previous review (ex cleaning up comments) hold true here. I just didn't bother writing them.

TheLegendaryGanjoo committed:
combined redundant functions + removed comments

Andallfor commented on Assets/Code/data classes/visibilty.cs:
```diff
@@ -31,51 +31,51 @@ public raycastInfo(bool hit, List<planet> HitPlanets, List<satellite> HitSats, L
-  //double.max
-
-  public static raycastInfo raycastHit(position p1, position p2, raycastParameters flags, int defaultRadius)
+  /// <summary> This is the main raycast function for our program and it takes in 5 arguments. The final argument if set true will return a list of all the hit objects, but if it is set to false the program will break after the first object is hit <summary> ///
```
Should be this format- `/// <summary> comment </summary>`

Andallfor commented on Assets/Code/data classes/visibilty.cs:
```diff
@@ -31,51 +31,51 @@ public raycastInfo(bool hit, List<planet> HitPlanets, List<satellite> HitSats, L
           {
-            hit = true;
-            hitPlanets.Add(p);
+            if (returnAllHit == true)
```
== true is implied, can just write `if (returnAllHit)`

Andallfor commented on Assets/Code/data classes/visibilty.cs:
```diff
@@ -31,51 +31,51 @@ public raycastInfo(bool hit, List<planet> HitPlanets, List<satellite> HitSats, L
-            hitPlanets.Add(p);
+            if (returnAllHit == true)
+            {
+              hit = true;
```
the content of the if statement is repeated. take the stuff that is repeated and put it outside the loop.

TheLegendaryGanjoo committed:
Update visibilty.cs

Andallfor committed:
Merge branch 'main' into Visiblity

Andallfor reviewed (commented):
All good
